app/assets/javascripts/basetypes/Code.js

```diff
@@ -9,19 +9,19 @@ Code.prototype = Object.create(mongoose.SchemaType.prototype);
 Code.prototype.cast = (code) => {
   if (code != null) {
     // handles codes that have not yet been cast to a code and those that have already been cast to a code
-    if (code.code && (code.codeSystemOid || code.system)) {
+    if (code.code && code.system) {
       if (typeof code.code === 'undefined') {
         throw new Error(`Code: ${code} does not have a code`);
-      } else if (typeof code.codeSystemOid === 'undefined' && typeof code.system === 'undefined') {
+      } else if (typeof code.system === 'undefined' && typeof code.system === 'undefined') {
```

Comment:
This if would never be true because the statement on line 12 essentially checks the same thing.